goaviatrix/vendor_integration.go

```diff
@@ -140,3 +141,29 @@ func (c *Client) SyncFireNetFirewallManagerVendorConfig(ctx context.Context, fir
 
 	return c.PostAPIContext(ctx, params["action"], params, BasicCheck)
 }
+
+func (c *Client) EditFireNetFirewallVendorInfoWithPrivateKey(vendorInfo *VendorInfo) error {
+	params := map[string]string{
+		"action":           "edit_firenet_firewall_vendor_info",
+		"CID":              c.CID,
+		"vpc_id":           vendorInfo.VpcID,
+		"firewall_id":      vendorInfo.InstanceID,
+		"firewall_name":    vendorInfo.FirewallName,
+		"public_ip":        vendorInfo.PublicIP,
+		"firewall_vendor":  vendorInfo.VendorType,
+		"user":             vendorInfo.Username,
+		"private_key_file": vendorInfo.PrivateKeyFile,
+	}
+
+	var files []File
+
+	key := File{
+		ParamName:      "private_key_file",
+		UseFileContent: true,
+		FileName:       "key.pem", // fake name for key
+		FileContent:    vendorInfo.PrivateKeyFile,
+	}
+	files = append(files, key)
```

Comment:
Why do you set the 'private_key_file' parameter twice? Once as text and again as file

Reply:
good catch